proxies/sandbox/apiproxy/resources/jsc/SetErrorVars.js

```diff
@@ -0,0 +1,202 @@
+    }
+
+    // $.data.attributes.messageBatchReference
+    pointer = "/data/attributes/messageBatchReference"
+    if (typeof data.attributes.messageBatchReference === 'undefined') {
+      errors.push(missingError(pointer))
+    } else if (data.attributes.messageBatchReference === null || data.attributes.messageBatchReference === "") {
+      errors.push(nullError(pointer))
+    } else if (typeof data.attributes.messageBatchReference !== "string" || !uuidRegex.test(data.attributes.messageBatchReference)) {
+      errors.push(invalidError(pointer))
+    }
+
+    // $.data.attributes.messages
+    pointer = "/data/attributes/messages"
+    if (typeof data.attributes.messages === 'undefined') {
+      errors.push(missingError(pointer))
+    } else if (!Array.isArray(data.attributes.messages)) {
+      errors.push(invalidError(pointer))
+    } else if (data.attributes.messages === null || Object.keys(data.attributes.messages).length === 0) {
+      errors.push(tooFewItemsError(pointer))
+    } else {
+
+      // $.data.attributes.recipients.x
+      data.attributes.messages.forEach((message, index) => {
+        pointer = "/data/attributes/messages/" + index
+        if (typeof message === 'undefined') {
+          errors.push(missingError(pointer))
+        } else if (message === null || message === {}) {
+          errors.push(nullError(pointer))
+        } else {
+
+          // $.data.attributes.messages.x.messageReference
+          pointer = "/data/attributes/messages/" + index + "/messageReference"
+          if (typeof message.messageReference === 'undefined') {
+            errors.push(missingError(pointer))
+          } else if (message.messageReference === null || message.messageReference === "") {
+            errors.push(nullError(pointer))
+          } else if (typeof message.messageReference !== "string" || !uuidRegex.test(message.messageReference)) {
+            errors.push(invalidError(pointer))
+          } else if (seenMessages.indexOf(message.messageReference) !== -1) {
+            errors.push(duplicateError(pointer))
+          }
+          seenMessages.push(message.messageReference)
+        }
+
+        // $.data.attributes.messages.x.recipient
+        pointer = "/data/attributes/messages/" + index + "/recipient"
+        if (typeof message.recipient === 'undefined') {
+          errors.push(missingError(pointer))
+        } else if (typeof message.recipient !== 'object') {
+          errors.push(invalidError(pointer))
+        } else if (message.recipient === null) {
+          errors.push(nullError(pointer))
+        } else {
+
+          // $.data.attributes.messages.x.recipient.nhsNumber
+          pointer = "/data/attributes/messages/" + index + "/recipient/nhsNumber"
+          if (typeof message.recipient.nhsNumber === 'undefined') {
+            errors.push(missingError(pointer))
+          } else if (message.recipient.nhsNumber === null || message.recipient.nhsNumber === "") {
```

Comment:
I believe the logic here is invalid, the null error should only throw when its null - an empty string is a none null value and should instead throw the invalid error. This should be the same across all fields.
* undefined == missing error
* null = null error
* invalid value (or type) == invalid error